sbol_utilities/helper_functions.py

```diff
@@ -363,4 +363,21 @@ def is_circular(obj: Union[sbol3.Component, sbol3.LocalSubComponent, sbol3.Exter
     :param obj: design to be checked
     :return: true if circular
     """    
-    return any(n==sbol3.SO_CIRCULAR for n in obj.types)
\ No newline at end of file
+    return any(n==sbol3.SO_CIRCULAR for n in obj.types)
+ 
+def is_composite(obj):
+    def has_dna_type(o):
+        return any(tyto.SO.DNA == tyto.SO.get_uri_by_term(t) for t in o.types)
+
+    def has_assembly_plan(o):
+        if not hasattr(o, "generated_by"):
```

Comment:
I don't think this should be necessary, because every `Component` has a generated_by attribute, even if it is empty.